WebJobs.sln

```diff
@@ -53,8 +53,6 @@ Project("{6EC3EE1D-3C4E-46DD-8F32-0CC8E7565705}") = "FSharpFunctions", "test\Tes
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Benchmarks", "test\Benchmarks\Benchmarks.csproj", "{78F8086D-8313-477A-B24F-E475A690880A}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "TestChildProcess", "TestChildProcess\TestChildProcess.csproj", "{CEBC078A-9DD2-4558-829B-DBAE6BEA4264}"
```

Comment:
I had a temporarily skipped E2E test for process monitoring that was spinning up test processes. The test ran fine locally and also on CI when I RDP'ed in and ran from VS. But for some reason in actual CI runs, the test would fail. The child processes were getting killed prematurely by someone. So I've made some refactorings to make Process fully mockable, since the goal of the test was to test behaviors when processes were killed (e.g. language worker dies).